### ISSUE TITLE
ForgeEssentials is not buggy as Cauldron, but it's heavier on the CPU

### ISSUE THREAD

**Issue by Starless2001** [ForgeEssentials/ForgeEssentialsMain#1728]
ForgeEssentials is a lot better than in the sense that mods work as they should, including event handlers and all that (except PlayerEvent.PlayerLoggedInEvent that doesn't seem to fire, maybe because of the auth module). but on the other hand, it seems to be a bit heavier on the host. what modules should I disable to fix that? Also, are you planning on optimizing performance after the project gets out of beta?

**Comment by olee** [comment]
I think it might be instead that just **cauldron's general performance might be better than forge alone**.
ForgeEssentials was already designed that it does not cause any performance impact when it is just running in the background. It does not perform any actions in the background without user interaction.

Only when a player breaks a block for example, FE will just check the permissions and that's it.

TL;DR: FE does not really impact server performance (unless there's a bug somewhere).

**Comment by liach** [comment]
So to get high performance we should use fastcraft?

**Comment by Starless2001** [comment]
I see. so the cost for better performance is a lot of bugs (Cauldron instead of pure Forge). cauldron is a good idea and all but unfortunately it has too many bugs. Maybe Forge could improve their stuff. thanks for the clarification

**Comment by olee** [comment]
@Starless2001 the problem is neither forge nor cauldron. The problem is 1) too many mods and 2) too much bad code in many mods.

But if you have a server with pretty much standard specs, there should be no performance problems running a forge server.

**Comment by luacs1998** [comment]
The thing about cauldron and derivatives is that it makes changes to minecraft/forge/bukkit code, not just to allow interoperability between the 3, but also quite a fair bit of performance optimization.

When mods aren't aware of these edits, shit hits the fan. You could try fastcraft (**USE THE 1.22 TEST BUILDS**), though.
